signac_dashboard/dashboard.py

```diff
@@ -473,17 +496,22 @@ def prevent_caching(response):
 
         @dashboard.app.context_processor
         def injections():
-            session.setdefault(
-                "enabled_modules",
-                [i for i in range(len(self.modules)) if self.modules[i].enabled],
-            )
+            keyfunc = lambda m: m.context
+            grouped_modules = groupby(sorted(self.modules, key=keyfunc), key=keyfunc)
+            enabled_module_indices = dict()
+            for context_key, context_group in grouped_modules:
+                enabled_module_indices[context_key] = [
+                    i for i, m in enumerate(context_group) if m.enabled
+                ]
+            session.setdefault("enabled_module_indices", enabled_module_indices)
             return {
                 "APP_NAME": "signac-dashboard",
                 "APP_VERSION": __version__,
                 "PROJECT_NAME": self.project.config["project"],
                 "PROJECT_DIR": self.project.config["project_dir"],
                 "modules": self.modules,
-                "enabled_modules": session["enabled_modules"],
+                "modules_by_context": self.modules_by_context,
```

Comment:
Yup, I think `self.modules_by_context` can be private. We inject `module_assets` despite it being private as well.